code/Pagetypes/FreshdeskPage.php

```diff
@@ -0,0 +1,262 @@
+            DropdownField::create('priority', 'priority', $priorities, $currentPriority)
+        );
+
+        $actions = new FieldList(
+            FormAction::create('Filter', 'Filter')
+        );
+
+        $form = new Form($this, '', $fields, $actions);
+        $form->setTemplate('FilterForm');
+        $form->setFormMethod('get');
+
+        return $form;
+    }
+
+    /**
+     * Get tickets from in memory cache, otherwise put them there.
+     *
+     * @param mixed $currentMember
+     *
+     * @return array $_tickets
+     */
+    private function getTickets($currentMember)
+    {
+        if (!self::$_tickets) {
+            $freshdesk = \FreshdeskAPI::create();
+            $result = $freshdesk->APICall('GET', FRESHDESK_API_BASEURL, '/api/v2/tickets?email='.urlencode($currentMember->Email));
+
+            $tickets = [];
+            if ($result && '200' == $result->getStatusCode()) {
+                $tickets = json_decode($result->getBody()->getContents(), true);
+            }
+            self::$_tickets = $tickets;
+        }
+
+        return self::$_tickets;
+    }
+
+    /**
+     * Filters tickets for a user based on product ID or other filter.
+     *
+     * @param array $tickets,  String $productID, Array $filter
+     * @param mixed $productID
+     * @param mixed $filter
+     *
+     * @return array $tickets
+     */
+    private function freshdeskTicketFilter($tickets, $productID = false, $filter)
+    {
+        // fast return if nothing to filter on
+        if (!$productID && !$filter) {
+            return $tickets;
+        }
+
+        foreach ($tickets as $key => $val) {
+            if ($productID && $val['product_id'] != $productID) {
+                unset($tickets[$key]);
+                continue;
+            }
+
+            if (empty($filter) || 0 == count($filter)) {
+                continue;
+            }
+
+            foreach ($filter as $filterKey => $filterVal) {
+                $doFilter = $this->doFilter($val, $filterKey, $filterVal);
+                if ($doFilter) {
+                    unset($tickets[$key]);
+                }
+            }
+        }
+
+        return $tickets;
+    }
+
+    /**
+     * Return true or false, depending on if a ticket should be removed based on a match to a filter.
+     *
+     * @param array $ticket,   String $filterKey, String $filterVal
+     * @param mixed $filterKey
+     * @param mixed $filterVal
+     *
+     * @return bool
+     */
+    private function doFilter($ticket, $filterKey, $filterVal)
+    {
+        if ($ticket[$filterKey] == (int) $filterVal) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Converts the numeric values in a ticket to something that makes sense to a human. Mappings defined as statics on this class.
+     *
+     * @param array $tickets
+     *
+     * @return array $formattedTickets
+     */
+    private function humanReadable($tickets)
```

Comment:
This mapping would most logically fit inside the API, so that Controller here can operate on the model (instead of int(2), we want FreshdeskTicketStatus(2) or whatever the model is.